### PR TITLE
etcd: etcd-backup.sh

### PR DESCRIPTION
**Pull request by ilixiaocui**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: etcd-backup.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# 备份路径
+backupPath=/etc/curve/etcd
+backupnum=5
+
+# 获取leader etcd
+leader_etcd=`curve_ops_tool etcd-status | grep "current etcd:" | awk -F" " '{print $3}'`
+if [ -z "$leader_etcd"]
+then
+    echo "get etcd leader fail"
+    exit 0
```
Comment (wu-hanqing) [inline etcd-backup.sh:12]: 出错退出可以exit 1吧？

File: etcd-backup.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+fi
+
+# 备份leader的数据库
+sudo etcdctl --endpoints ${leader_etcd} snapshot save ${backupPath}/snapshot_`date +%Y%m%d%H%M`.db
+if [ $? -ne 0 ]
+then
+    echo "backup etcd fail"
+    exit 0
```
Comment (wu-hanqing) [inline etcd-backup.sh:20]: 同上

File: etcd-backup.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+fi
+
+# 超过backupnum的备份数据删除
```
Comment (wu-hanqing) [inline etcd-backup.sh:23]: 这里应该可以用 `find ${backupPath} -name "snapshot_*" -mtime +7 | xargs rm`

**Review by wu-hanqing — CHANGES_REQUESTED** [review]

File: etcd-backup.sh [hunk 4]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+# 备份路径
+backupPath=/etc/curve/etcd
+backupnum=5
+
+# 获取leader etcd
+leader_etcd=`curve_ops_tool etcd-status | grep "current etcd:" | awk -F" " '{print $3}'`
```
Comment (SeanHai) [inline etcd-backup.sh:8]: awk -F" " (默认分隔符，可以省略），加上也可以吧

Comment (ilixiaocui) [reply etcd-backup.sh]: done

File: etcd-backup.sh (hunk 1 above)
Comment (ilixiaocui) [reply etcd-backup.sh]: done

File: etcd-backup.sh (hunk 3 above)
Comment (ilixiaocui) [reply etcd-backup.sh]: 👍

File: etcd-backup.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+# 备份路径
```
Comment (aspirer) [inline etcd-backup.sh:3]: 如果要merge到主库，还是改成英文比较合适。

**Comment by aspirer** [comment]
恢复也写个脚本？主要是那些参数看着还挺复杂的，临时准备不知道好不好找。

**Comment by opencurveadmin** [comment]
recheck

**Comment by opencurveadmin** [comment]
recheck

**Pull request closed without merging**